sbin/repair_services.sh

```diff
@@ -1,36 +1,122 @@
 #!/usr/bin/env bash
 # shellcheck enable=all
 
-#### This script repairs core services that are not currently running
+#### This script repairs core services that are not currently active
 
-repair_services() {
-    services=(
-        "nginx"
-        "apache2"
-        "postgresql"
-        "nullsmtpd"
-        "submitty_websocket_server"
-        "submitty_autograding_worker"
-        "submitty_autograding_shipper"
-        "submitty_daemon_jobs_handler"
-    )
-
-    for service in "${services[@]}"; do
-        if ! sudo systemctl is-active --quiet "${service}"; then
-            today=$(date +%Y%m%d)
-            log_file="/var/log/services/${today}.txt"
-            last_status=$(sudo systemctl status "${service}")
-
-            if [[ ! -f "${log_file}" ]]; then
-                sudo touch "${log_file}"
+declare -A services=(
+    ["nginx"]="Nginx Server"
+    ["apache2"]="Web Server"
+    ["postgresql"]="Database Server"
+    ["nullsmtpd"]="Mail Server"
+    ["autograding"]="Autograding Shipper/Workers"
+    ["submitty_websocket_server"]="WebSocket Server"
+    ["submitty_daemon_jobs_handler"]="Daemon Jobs Handler"
+)
+
+log_service_restart() {
+    local service="$1"
+    local message="$2"
+    local last_status="$3"
+
+    local today
+    today=$(date +%Y%m%d)
+    local timestamp
+    timestamp=$(date "+%Y-%m-%d:%H:%M:%S")
+    local log_file="/var/log/services/${today}.txt"
```

Comment:
Let's change the path of these logs to be "/var/local/submitty/logs/services/"
(this path might conflict with some other linux package now or in the future...)